src/term.rs

```diff
@@ -288,6 +288,36 @@ impl Term {
     pub fn rhs_mut(&mut self) -> Result<&mut Term, TermError> {
         if let Ok((_, rhs)) = self.unapp_mut() { Ok(rhs) } else { Err(NotApp) }
     }
+
+    /// Returns `true` if and only if the lambda term is a
+    /// [supercombinator](https://en.wikipedia.org/wiki/Supercombinator).
+    ///
+    /// # Example
+    /// ```
+    /// use lambda_calculus::term::*;
+    ///
+    /// let term1 = abs(app(Var(1), abs(Var(1)))); // λ 1 (λ 2)
+    /// let term2 = app(abs(Var(2)), abs(Var(1))); // (λ 2) (λ 1)
+    ///
+    /// assert_eq!(term1.is_supercombinator(), true);
+    /// assert_eq!(term2.is_supercombinator(), false);
+    /// ```
+    pub fn is_supercombinator(&self) -> bool {
+        let mut stack = Vec::new();
+        stack.push((0, self));
```

Comment:
it will probably never be an issue in practice, but `0` is inferred as `i32` here; it would be a good idea to annotate it as `0usize`